### PR TITLE
Fix #975 - Require email address to match domain

### LINKED ISSUE #975: Require email address to match domain
When registering for an account, it would be great to restrict that ability to only users of a certain email domain.

### PR DESCRIPTION
**Pull request by Ribeiro**
Please, take a look  =)

### CODE REVIEW

File: packages/rocketchat-lib/settings/server/startup.coffee [hunk 1]
```diff
@@ -38,6 +38,9 @@ RocketChat.settings.addGroup 'General'
 RocketChat.settings.add 'Site_Name', 'Rocket.Chat', { type: 'string', group: 'General', public: true }
 RocketChat.settings.add 'Allow_Invalid_SelfSigned_Certs', false, { type: 'boolean', group: 'General' }
 RocketChat.settings.add 'Disable_Favorite_Rooms', false, { type: 'boolean', group: 'General' }
+RocketChat.settings.add 'Allowed_Domains', 'gmail.com;hotmail.com', { type: 'string', group: 'General'}
+RocketChat.settings.add 'At_Sign', '@', { type: 'string', group: 'General'}
+RocketChat.settings.add 'Values_Separator', ';', { type: 'string', group: 'General'}
```
Comment (rodrigok) [inline packages/rocketchat-lib/settings/server/startup.coffee]: I think that this 2 options (At_Sign and Values_Separator) should not be settings

File: server/lib/accounts.coffee [hunk 2]
```diff
@@ -19,6 +19,14 @@ Accounts.onCreateUser (options, user) ->
 	# console.log 'options ->',JSON.stringify options, null, '  '
 	# console.log 'user ->',JSON.stringify user, null, '  '
 
+	userEmail = user.emails[0].address;
```
Comment (rodrigok) [inline server/lib/accounts.coffee]: Should exists an setting to enable/disable the domain verification cuz there are installations that don't require email on signup

Comment (Ribeiro) [inline server/lib/accounts.coffee]: @rodrigok I didn´t know about that!

File: packages/rocketchat-lib/settings/server/startup.coffee (hunk 1 above)
Comment (Ribeiro) [inline packages/rocketchat-lib/settings/server/startup.coffee]: @rodrigok We can change this later, after defining the official domain´s whitelist  =)

**Comment by gmsecrieru** [comment]
@Ribeiro I have created a new branch with a simple code change that should help you get going.

Steps:
1. Sync your code with [domain-whistelist-POC](https://github.com/RocketChat/Rocket.Chat/compare/domain-whistelist-POC) branch
2. `meteor`
3. Log in with your Admin user
4. Click your avatar -> Administration -> Accounts -> Look for "Account_AllowedDomainsList" and type `gmail.com, yahoo.com`
5. Save your changes
6. *IMPORTANT* restart `meteor` (should not be needed when #913 is implemented)

You should see the following:

*Invalid e-mail address*
screen shot 2015-10-09 at 17 19 12

*Valid e-mail address* (e.g. foo@gmail.com)
screen shot 2015-10-09 at 17 19 27

Could you please sync your fork with mentioned branch and use that as the foundation for your PR?

/cc @rodrigok

File: packages/rocketchat-lib/settings/server/startup.coffee (hunk 1 above)
Comment (rodrigok) [inline packages/rocketchat-lib/settings/server/startup.coffee]: What do you mean with `oficial domain's whitelist`?

File: server/lib/accounts.coffee (hunk 2 above)
Comment (rodrigok) [inline server/lib/accounts.coffee]: Can you explain more?

**Comment by gmsecrieru** [comment]
@Ribeiro Did you get the chance to look into this? Let us know should you have any questions.

**Comment by geekgonecrazy** [comment]
Is this pretty much replaced ny #1182 ?

**Comment by gmsecrieru** [comment]
Yes!

**Comment by intositeme** [comment]
how do you restart meteor running in docker?

**Comment by geekgonecrazy** [comment]
@intositeme you do docker stop {container_name} then start the container again. docker start {container_name}

**Pull request closed without merging**